ci/ci/ci.py

```diff
@@ -395,6 +396,18 @@ async def batch_callback_handler(request):
             for wb in watched_branches:
                 if wb.branch.short_str() == target_branch:
                     log.info(f'watched_branch {wb.branch.short_str()} notify batch changed')
+
+                    if 'test' in params and params['complete']:
+                        assert 'deploy' not in params
+                        assert 'dev' not in params
+                        db: Database = app['db']
+                        namespaces = json.loads(params['namespaces'])
+                        await db.execute_update(
+                            f'''
+DELETE FROM active_namespaces
+WHERE {" OR ".join("namespace=" + ns for ns in namespaces)}'''
```

Comment:
I'd feel more comfortable here if we used a query string with `%s` and gave the namespace names as query args.

Reply:
oops, totally